test/e2e/manifest_test.go

```diff
@@ -16,6 +17,50 @@ import (
 	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+// validateManifestHasAllArchs checks that the specified manifest has all
+// the archs in `imageListInstance`
+func validateManifestHasAllArchs(path string) error {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	var result map[string]interface{}
```

Comment:
Absolutely non-blocking: Defining a function-private type might be shorter and easier to read:
```go
var result struct {
   Manifests []struct {
      Platform struct {
         Architecture string
      }
   }
}
```
or something like that.

OTOH this works just fine, and passes tests as is.

Reply:
your suggestion is better, fixed now